hw/bsp/nrf52840pdk/src/hal_bsp.c

```diff
@@ -200,6 +204,13 @@ hal_bsp_init(void)
     /* Make sure system clocks have started */
     hal_system_clock_start();
 
+    hal_trng_init();
+
+    rc = os_dev_create(&os_bsp_trng.dev, "nrf_rng",
```

Comment:
wouldn't it be better to call it just `trng`?

Reply:
agree, but how about this: we may have both `hal_trng` and `hw/drivers/trng`. on platforms with TRNG in SoC it can be available as HAL and also as os_dev called `hal_trng`. obviously driver will use HAL implementaion instead of duplicating code as it is now.

Reply:
...or it can be even called `trng` as suggested - like a main TRNG on BSP ;-)